src/App.tsx

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useReducer } from 'react'
+import Grid from './components/grid'
+import { CellSize, GameReducer, reset } from './env'
+import { GameContext, GameDispatchContext } from './env/context'
+import Header from './components/header'
+import './App.css'
+import Footer from './components/footer'
+
+function App() {
+    const [state, dispatch] = useReducer(GameReducer, reset())
+
+    useEffect(() => {
```

Comment:
@aronluigi these 2 `useEffect` calls include game logic. these can be moved to `env/index.ts` where all the other game logic is collected.

Reply:
This will not work because React Hook "useEffect" cannot be called at the top level. React Hooks must be called in a React function component or a custom React Hook function. If I'm not mistaken, it could be an antipattern.

Reply:
@aronluigi you can make it work with a custom hook:

env/index.ts:
```
function useGameReducer() {
   return useReducer(GameReducer, reset())
}

export function useGameStep() {
   const [state, dispatch] = useGameReducer()
   useEffect(() => {
       dispatch({ type: 'step' })
    }, [state.stepsLeft])
}

export function useGameRoll() {
   ....
}

```
App.tsx:

```
function App() {
   useGameStep()
   useGameRoll()
   return ...
}
```